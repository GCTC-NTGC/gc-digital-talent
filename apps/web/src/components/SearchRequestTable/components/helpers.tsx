import React from "react";
import { IntlShape } from "react-intl";

import { Link, Pill, Spoiler } from "@gc-digital-talent/ui";
import { formatDate, parseDateTimeUtc } from "@gc-digital-talent/date-helpers";
import { notEmpty } from "@gc-digital-talent/helpers";
import {
  commonMessages,
  getPoolCandidateSearchStatus,
} from "@gc-digital-talent/i18n";

import {
  Classification,
  Maybe,
  PoolCandidateSearchRequest,
  PoolCandidateSearchStatus,
  Scalars,
} from "~/api/generated";

export function classificationAccessor(
  classifications: Maybe<Maybe<Classification>[]> | undefined,
) {
  return classifications
    ?.filter(notEmpty)
    ?.map((c) => `${c.group}-0${c.level}`)
    ?.join(", ");
}

export function classificationsCell(
  classifications: Maybe<Maybe<Classification>[] | undefined> | undefined,
  intl: IntlShape,
) {
  const filteredClassifications = classifications
    ? classifications.filter(notEmpty)
    : [];
  const pillsArray = filteredClassifications.map((classification) => {
    return (
      <Pill
        key={`${classification.group}-0${classification.level}`}
        color="primary"
        mode="outline"
      >
        {`${classification.group}-0${classification.level}`}
      </Pill>
    );
  });
  return pillsArray.length > 0 ? (
    <span>{pillsArray}</span>
  ) : (
    intl.formatMessage(commonMessages.notProvided)
  );
}

export function dateCell(date: Maybe<Scalars["DateTime"]>, intl: IntlShape) {
  return date ? (
    <span>
      {formatDate({
        date: parseDateTimeUtc(date),
        formatString: "PPP p",
        intl,
      })}
    </span>
  ) : null;
}

export const notesCell = (
  searchRequest: PoolCandidateSearchRequest,
  intl: IntlShape,
) =>
  searchRequest?.adminNotes ? (
    <Spoiler
      text={searchRequest.adminNotes}
      linkSuffix={intl.formatMessage(
        {
          defaultMessage: "notes for {name}",
          id: "6eih3b",
          description:
            "Link text suffix to read more notes for a search request",
        },
        {
          name: searchRequest.jobTitle,
        },
      )}
    />
  ) : null;

export const detailsCell = (
  searchRequest: PoolCandidateSearchRequest,
  intl: IntlShape,
) =>
  searchRequest?.additionalComments ? (
    <Spoiler
      text={searchRequest.additionalComments}
      linkSuffix={intl.formatMessage(
        {
          defaultMessage: "details for {name}",
          id: "sl1kbp",
          description:
            "Link text suffix to read more details for a search request",
        },
        {
          name: searchRequest.jobTitle,
        },
      )}
    />
  ) : null;

export const statusCell = (
  status: PoolCandidateSearchStatus | null | undefined,
  intl: IntlShape,
) =>
  status
    ? intl.formatMessage(getPoolCandidateSearchStatus(status as string))
    : "";

export function viewCell(url: string, label: Maybe<string>, intl: IntlShape) {
  return (
    <Link href={url} color="black">
      {label ||
        intl.formatMessage({
          defaultMessage: "No name provided",
          id: "L9Ked5",
          description: "Fallback for team display name value",
        })}
    </Link>
  );
}
