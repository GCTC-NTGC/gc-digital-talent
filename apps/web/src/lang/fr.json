{
  "+JIfBy": {
    "defaultMessage": "Dans quel ministère ou organisme les apprentis travailleront-ils?",
    "description": "Learn more dialog question nine heading"
  },
  "+cn9MN": {
    "defaultMessage": "Les personnes qui se voient offrir des postes avec la possibilité de travailler à domicile doivent avoir un accès Internet adéquat pour le travail à distance.",
    "description": "IAP Requirement list item six"
  },
  "+do6jV": {
    "defaultMessage": "Voir les critères d’admissibilité",
    "description": "Button text for program eligibility criteria"
  },
  "/5tYua": {
    "defaultMessage": "Fournissez vos renseignements",
    "description": "How it works, step 2 heading"
  },
  "/7NWbF": {
    "defaultMessage": "Nous vous remercions de votre intérêt dans ce programme. Nous attendons de vos nouvelles avec impatience!",
    "description": "Third paragraph for apply now dialog"
  },
  "/ar49+": {
    "defaultMessage": "S’engager à apprendre tout au long du programme, tant au travail qu’en suivant un programme de formation en ligne",
    "description": "IAP Requirement list item five"
  },
  "1M3LT6": {
    "defaultMessage": "Un processus de demande en ligne sera bientôt disponible afin de permettre aux gens d’exprimer leur intérêt. Entre-temps, nous vous encourageons à envoyer votre curriculum vitae et votre lettre de présentation par courriel, à edsc.pda-iap.esdc@hrsdc-rhdcc.gc.ca.",
    "description": "Learn more dialog question eleven paragraph one"
  },
  "27FjxB": {
    "defaultMessage": "Avoir une passion ou un intérêt pour la technologie de l’information (TI), avec cette passion ou cet intérêt démontrés par une expérience personnelle, bénévole, communautaire ou professionnelle.",
    "description": "IAP Requirement list item four"
  },
  "404R1N": {
    "defaultMessage": "Ce programme a été pour moi une occasion de changer ma vie et je vois <b>un meilleur avenir</b>.",
    "description": "testimonial number one"
  },
  "436DA5": {
    "defaultMessage": "Croissance",
    "description": "Talent portal strategy item 2 heading"
  },
  "4p0QdF": {
    "defaultMessage": "Fermer",
    "description": "Button text used to close an open modal"
  },
  "6l9YH+": {
    "defaultMessage": "Les demandeurs doivent répondre aux exigences suivantes :",
    "description": "Heading for the applicant requirements dialog"
  },
  "81iywD": {
    "defaultMessage": "On vous demandera de confirmer les renseignements que vous avez fournis.",
    "description": "How it works, step 3 content sentence 1"
  },
  "84rSVg": {
    "defaultMessage": "Permettre une croissance dans le cadre de sa portée de recrutement, en ciblant d’autres domaines professionnels à l’avenir.",
    "description": "Talent portal strategy item 2 content"
  },
  "8PhWBd": {
    "defaultMessage": "Lancement",
    "description": "Talent portal strategy item 4 heading"
  },
  "BSSYnh": {
    "defaultMessage": "Nous voulons vous informer qu’entre-temps, des mises à jour sont apportées à ce site qui permettront aux personnes autochtones qui souhaitent se joindre au Programme d’apprentissage en TI de présenter une demande en ligne.",
    "description": "Second paragraph for apply now dialog"
  },
  "CY+493": {
    "defaultMessage": "Pour en apprendre davantage",
    "description": "Button text to learn more about the program"
  },
  "CZGta2": {
    "defaultMessage": "Y a-t-il des frais pour participer au programme?",
    "description": "Learn more dialog question ten heading"
  },
  "CqLV19": {
    "defaultMessage": "À propos du programme",
    "description": "Program information section title"
  },
  "DBczOG": {
    "defaultMessage": "Stratégie",
    "description": "Heading for strategy for the indigenous talent portal"
  },
  "DgMIfz": {
    "defaultMessage": "Le Programme d’apprentissage en TI vous convient-il?",
    "description": "Application box heading part one"
  },
  "DvmNR7": {
    "defaultMessage": "Inscrivez-vous dès maintenant",
    "description": "Button text to apply for program"
  },
  "Dzyk1q": {
    "defaultMessage": "En collaboration avec le Programme d’apprentissage en TI pour les peuples autochtones, le Portail des talents autochtones commencera par mettre l’accent sur les talents en TI et en technologie, ce qui permettra :",
    "description": "Description for strategy for the indigenous talent portal"
  },
  "EqVMCE": {
    "defaultMessage": "Veuillez envoyer votre curriculum vitae et votre lettre de présentation expliquant votre passion pour la TI et pourquoi vous souhaitez vous joindre au programme, à : <mailLink>edsc.pda-iap.esdc@hrsdc-rhdcc.gc.ca</mailLink>. Un membre de l’équipe communiquera avec vous dans 3 à 5 jours ouvrables.",
    "description": "First paragraph for apply now dialog"
  },
  "GnmmgR": {
    "defaultMessage": "Une fois votre demande présenté, un membre de l’équipe communiquera avec vous dans les 3 à 5 jours ouvrables qui suivent.",
    "description": "How it works, step 3 content sentence 2"
  },
  "qZvV7b": {
    "defaultMessage": "Programme d’apprentissage en TI pour les peuples autochtones. Postulez dès aujourd’hui pour entamer votre parcours de carrière en TI.",
    "description": "Homepage title for Indigenous Apprenticeship Program"
  },
  "Ilot3b": {
    "defaultMessage": "Que se passe-t-il après lorsqu’une personne achève le programme de 24 mois?",
    "description": "Learn more dialog question three heading"
  },
  "KdXvmY": {
    "defaultMessage": "Comment puis-je savoir si ce programme est fait pour moi?",
    "description": "Learn more dialog question seven heading"
  },
  "L9yjr3": {
    "defaultMessage": "Découvrez-en davantage sur le Programme d’apprentissage en TI pour les peuples autochtones du gouvernement du Canada",
    "description": "Heading for the Learn more dialog"
  },
  "LGSw+f": {
    "defaultMessage": "Tous les apprentis seront appuyés par des pairs et des mentors soigneusement sélectionnés. Les apprentis seront également invités à se joindre à un Réseau d’apprentissage pour les Autochtones, où ils auront des occasions de communiquer avec d’autres employés autochtones du gouvernement du Canada et avec d’autres apprentis autochtones qui participent à ce programme.",
    "description": "Learn more dialog question two paragraph two"
  },
  "MgLLHd": {
    "defaultMessage": "Demande élevée",
    "description": "Talent portal strategy item 1 heading"
  },
  "NaF4Iu": {
    "defaultMessage": "Inscrivez-vous dès aujourd’hui!",
    "description": "Application box heading part two"
  },
  "O0z6Ym": {
    "defaultMessage": "À qui ce programme est-il destiné?",
    "description": "Heading for section about who the program is for"
  },
  "OcIvN1": {
    "defaultMessage": "Une fois le programme terminé, on espère que les apprentis envisageront de rester au gouvernement du Canada.",
    "description": "Learn more dialog question three paragraph one"
  },
  "PvH5lJ": {
    "defaultMessage": "Ce que nous entendons",
    "description": "Heading for the quotes sections"
  },
  "QkZ7Oy": {
    "defaultMessage": "Comment une personne peut-elle présenter une demande pour devenir apprentie?",
    "description": "Learn more dialog question eleven heading"
  },
  "Qt9Q8k": {
    "defaultMessage": "Le programme dure combien de temps?",
    "description": "Learn more dialog question two heading"
  },
  "S/PXt/": {
    "defaultMessage": "Des possibilités existent au sein d’un grand nombre de ministères et d’organismes du gouvernement du Canada. Le programme tentera de jumeler les apprentis à des possibilités qui correspondent le mieux à leurs intérêts.",
    "description": "Learn more dialog question nine paragraph one"
  },
  "SnnuD+": {
    "defaultMessage": "Nous vous inviterons à créer un profil qui sera sauvegardé et présenté comme étant votre demande.",
    "description": "How it works, step 2 content sentence 2"
  },
  "Srs7a4": {
    "defaultMessage": "Passer au contenu principal",
    "description": "Assistive technology skip link"
  },
  "TFeQL2": {
    "defaultMessage": "Accueil",
    "description": "Link to the homepage for indigenous apprenticeship program."
  },
  "U8bLT7": {
    "defaultMessage": "Comment cela fonctionnera-t-il?",
    "description": "heading for how the indigenous talent portal will work"
  },
  "UDHGGA": {
    "defaultMessage": "Bientôt, les postulants pourront présenter leur demande à l’aide d’un outil interactif en ligne qui sera disponible à partir du présent site Web. Voici ce sur quoi nous travaillons :",
    "description": "Description of how the indigenous talent portal will work"
  },
  "UQlyXu": {
    "defaultMessage": "Les apprentis feront l’expérience d’un apprentissage au travail et pendant la formation en ligne dans un ou plusieurs domaines de la technologie de l’information. On s’attend à ce que les apprentis démontrent leur intérêt et leur passion pour la technologie en étant prêts à observer, à apprendre et à pratiquer de nouvelles compétences qui sont développées au cours du programme. En moyenne, un apprenti passera quatre jours par semaine en apprentissage pratique, tandis que le cinquième jour de la semaine sera consacré au perfectionnement personnel et professionnel par la formation en ligne et par d’autres possibilités de formation et de perfectionnement.",
    "description": "Learn more dialog question eight paragraph one"
  },
  "WDRWcu": {
    "defaultMessage": "Dois-je déménager à Ottawa ou à un endroit précis pour participer à ce programme?",
    "description": "Learn more dialog question six heading"
  },
  "XS21o1": {
    "defaultMessage": "Bien sûr! En tant qu’employés du gouvernement du Canada, tous les apprentis auront droit à des congés annuels, à des congés de maladie et à des congés familiaux. Après six mois, les apprentis auront également accès à une assurance médicale et dentaire pour eux-mêmes et leurs personnes à charge, en plus de devenir membres du régime de pension de retraite de la fonction publique fédérale.",
    "description": "Learn more dialog question five paragraph one"
  },
  "Xhfkfg": {
    "defaultMessage": "Répondre à la forte demande de talents autochtones en TI.",
    "description": "Talent portal strategy item 1 content"
  },
  "YGM/D5": {
    "defaultMessage": "Le programme dure 24 mois et les apprentis ont accès à un soutien pendant leur période de participation au programme.",
    "description": "Learn more dialog question two paragraph one"
  },
  "b09U1u": {
    "defaultMessage": "À la fin de leurs 24 mois, les apprentis auront des attestations et des compétences en demande, ainsi que la confiance nécessaire pour contribuer à la main-d’œuvre numérique du Canada, tant à l’intérieur qu’à l’extérieur de la fonction publique fédérale.",
    "description": "First paragraph what will you learn at the program"
  },
  "cNXORI": {
    "defaultMessage": "Être âgé(e) d’au moins 16 ans",
    "description": "IAP Requirement list item three"
  },
  "cYq1Dp": {
    "defaultMessage": "Les apprentis qui participent au programme disent que ça « change une vie », que ça représente « une occasion d’avoir une meilleure vie grâce à la technologie » et qu’« il n’y a aucun obstacle à la réussite de ce programme ».",
    "description": "Third paragraph about the program"
  },
  "ce892L": {
    "defaultMessage": "Inscrivez-vous dès maintenant",
    "description": "Heading for the apply now dialog"
  },
  "dDHy5d": {
    "defaultMessage": "Les apprentis suivent un programme structuré de 24 mois comprenant un mélange d’apprentissages en cours d’emploi et de formation formelle.",
    "description": "First paragraph what will you learn at the program"
  },
  "dgG8md": {
    "defaultMessage": "Quelles sont les exigences en matière d’études – puis-je présenter une demande si je n’ai pas de diplôme collégial ou universitaire?",
    "description": "Learn more dialog question twelve heading"
  },
  "fF4Ex+": {
    "defaultMessage": "Le Portail des talents autochtones a été créé pour la communauté autochtone par la communauté autochtone.",
    "description": "Talent portal information sentence 1"
  },
  "fvsYkj": {
    "defaultMessage": "Qu’est-ce que j’apprendrai dans le cadre de ce programme d’apprentissage?",
    "description": "What applicants will learn sections heading"
  },
  "hSmKjD": {
    "defaultMessage": "Absolument! Votre temps est précieux. Les apprentis recevront une rémunération et des avantages sociaux à titre d’employés du gouvernement du Canada. Au départ, les apprenti(e)s gagneront entre 29 $ et 33 $ de l’heure. Le salaire exact est déterminé au moment de l’embauche, et dépend de la convention collective en vigueur au sein de l’organisation qui les embauche. Les apprentis seront employés à temps plein, pour un total de 37,5 heures par semaine.",
    "description": "Learn more dialog question four paragraph one"
  },
  "hdBTPI": {
    "defaultMessage": "Tout l’équipement informatique requis (ordinateur portable, souris, écran, téléphone) et le matériel de formation seront fournis gratuitement. Si une personne choisit de travailler à la maison, le coût de l’accès à Internet relèvera de l’apprenti.",
    "description": "Learn more dialog question ten paragraph one"
  },
  "jA4kuU": {
    "defaultMessage": "Les apprentis seront-ils payés?",
    "description": "Learn more dialog question four heading"
  },
  "lIHNov": {
    "defaultMessage": "Évaluer",
    "description": "Talent portal strategy item 3 heading"
  },
  "loDwKe": {
    "defaultMessage": "À propos du Portail des talents autochtones",
    "description": "Talent Portal information heading"
  },
  "mfBsJY": {
    "defaultMessage": "Que devront faire les apprentis dans le cadre du programme?",
    "description": "Learn more dialog question eight heading"
  },
  "osGGIt": {
    "defaultMessage": "Programme d’apprentissage en TI pour les personnes autochtones + Portail des talents autochtones",
    "description": "heading for indigenous talent portal section"
  },
  "p19YJ2": {
    "defaultMessage": "Présentez une demande dès aujourd’hui pour entreprendre une carrière en technologie de l’information.",
    "description": "Application box content"
  },
  "p2sBh3": {
    "defaultMessage": "Qui peut présenter une demande pour devenir apprenti dans le cadre du Programme d’apprentissage en TI pour les peuples autochtones du gouvernement du Canada?",
    "description": "Learn more dialog question one heading"
  },
  "pWoAv0": {
    "defaultMessage": "Le Programme d’apprentissage en TI pour les personnes autochtones est une initiative du gouvernement du Canada visant particulièrement les Premières Nations, les Inuits et les Métis. C’est un chemin vers un emploi dans la fonction publique fédérale pour les personnes autochtones qui ont une passion pour la technologie de l’information (TI).",
    "description": "First paragraph about the program"
  },
  "pkFLpT": {
    "defaultMessage": "Si vous avez un intérêt et une passion pour la technologie et que vous vous identifiez comme Autochtone, ce programme est pour vous.",
    "description": "Learn more dialog question seven paragraph one"
  },
  "q5FQbu": {
    "defaultMessage": "À venir!",
    "description": "Heading for a coming soon section"
  },
  "rQ61eh": {
    "defaultMessage": "Permettre la collecte des données et de la rétroaction pour les mettre à profit afin d’améliorer le service.",
    "description": "Talent portal strategy item 3 content"
  },
  "rz21yp": {
    "defaultMessage": "Les apprentis auront-ils droit à des prestations?",
    "description": "Learn more dialog question five heading"
  },
  "sycoGU": {
    "defaultMessage": "Pas nécessairement. La plupart des possibilités d’apprentissage peuvent être réalisées en travaillant de la maison, pourvu qu’il y ait une connexion à Internet. Certains postes ont des exigences propres à l’emplacement. Nous accordons une importance à la famille et à la communauté, et nous encourageons les gens à se joindre à notre famille sans avoir à quitter la leur, et à faire partie de notre communauté tout en restant dans la leur. Parlez à votre gestionnaire d’embauche de votre situation ou de vos préférences, et nous ferons de notre mieux pour travailler avec vous afin de trouver une solution appropriée.",
    "description": "Learn more dialog question six paragraph one"
  },
  "wHWE3M": {
    "defaultMessage": "Il s’agit d’une plateforme conçue pour offrir des possibilités d’emploi aux Autochtones d’une manière qui reconnaît et met en valeur leurs talents, leurs idées, leurs compétences et leur passion uniques.",
    "description": "Talent portal information sentence 2"
  },
  "wgIThY": {
    "defaultMessage": "Avoir un diplôme d’études secondaires ou l’équivalent (formation générale)",
    "description": "IAP Requirement list item two"
  },
  "wqwPhL": {
    "defaultMessage": "En valorisant et en misant sur le potentiel d’une personne plutôt que sur son niveau de scolarité, le programme élimine l’un des plus grands obstacles qui existe en matière d’emploi dans l’économie numérique. Le programme a été élaboré par, avec et pour des personnes autochtones à travers le Canada. Il intègre les préférences et les besoins des apprenants autochtones tout en reconnaissant l’importance de la communauté.",
    "description": "Second paragraph about the program"
  },
  "yZMQ6j": {
    "defaultMessage": "Nous voulons en apprendre davantage sur vous et sur votre intérêt ou votre passion dans le domaine de la TI!",
    "description": "How it works, step 2 content sentence 1"
  },
  "zlYw3Z": {
    "defaultMessage": "Présentez votre profil en tant que demande",
    "description": "How it works, step 3 heading"
  },
  "1B4niz": {
    "defaultMessage": "Le Programme a été conçu pour répondre aux efforts de réconciliation et pour bâtir une relation renouvelée qui est fondée sur les droits, le respect, la collaboration et le partenariat avec les peuples autochtones.",
    "description": "How it works, step 1 content paragraph 1"
  },
  "jyQi0m": {
    "defaultMessage": "Si vous êtes membre d’une Première Nation, Inuk ou Métis, et que vous vivez au Canada, vous pouvez demander à devenir un apprentis dans le cadre de ce programme.",
    "description": "Learn more dialog question on paragraph one"
  },
  "nzcwrW": {
    "defaultMessage": "Les apprentis seront jumelés à un pair afin de faciliter l’observation en situation de travail et le travail supervisé, et on leur attribuera un mentor qui saura leur fournir des conseils d’une personne expérimentée et une orientation au fil de la durée du programme.",
    "description": "First paragraph what will you learn at the program"
  },
  "pPOGwF": {
    "defaultMessage": "Ils sont membres des Premières Nations (avec ou sans statut), Inuit ou Métis ",
    "description": "IAP Requirement list item one"
  },
  "49UIne": {
    "defaultMessage": "Apprenti en TI du Gouvernement du Canada",
    "description": "author of testimonial number one"
  },
  "74Sg+4": {
    "defaultMessage": "L’exigence minimale en matière d’études est un diplôme d’études générales ou secondaires. Il n’est pas nécessaire d’avoir fait des études collégiales ou universitaires.",
    "description": "Learn more dialog question twelve paragraph one"
  },
  "khChKa": {
    "defaultMessage": "Le Programme s’adresse aux Premières Nations, aux Inuits et aux Métis. Si vous êtes un membre des Premières Nations, un Inuit ou un Métis et que vous avez une passion pour la technologie, ce programme est pour vous!",
    "description": "First paragraph about who the program is for"
  },
  "1FM1VL": {
    "defaultMessage": "Si vous n’êtes pas certain(e) que ce programme vous convient, veuillez <mailLink>communiquer avec nous</mailLink> et un membre de notre équipe se fera un plaisir de vous rencontrer pour répondre à vos questions.",
    "description": "Second paragraph about who the program is for"
  },
  "1pFMSH": {
    "defaultMessage": "Remplir la Formulaire d’autodéclaration à l’intention des peuples autochtones",
    "description": "How it works, step 1 heading"
  },
  "uJ1rk3": {
    "defaultMessage": "Il existe trois groupes distincts de peuples autochtones qui sont reconnus par la Constitution canadienne. On vous demandera de confirmer au(x)quel(s) de vous appartenez par l’entremise de la Formulaire d’autodéclaration à l’intention des peuples autochtones.",
    "description": "How it works, step 1 content paragraph 2"
  },
  "0i34ZZ": {
    "defaultMessage": "Vise à lancer le programme au début de 2023.",
    "description": "Talent portal strategy item 4 content"
  },
  "Oh1/Gc": {
    "defaultMessage": "Postulez dès aujourd'hui pour entamer votre parcours de carrière en TI.",
    "description": "Meta tag description for Indigenous Apprenticeship Program site"
  },
  "C5tUG2": {
    "defaultMessage": "Programme d’apprentissage destiné aux Autochtones",
    "description": "Title tag for Indigenous Apprenticeship Program site"
  },
  "t2Q+IG": {
    "defaultMessage": "Retourner au Français",
    "description": "Button text for returning to default language from Indigenous language"
  },
  "kiH9nz": {
    "defaultMessage": "Sélectionnez<hidden> une langue autochtone </hidden>",
    "description": "Button text displayed for Indigenous languages dropdown"
  },
  "uoX7ou": {
    "defaultMessage": "Cette page est accessible en langues autochtones",
    "description": "Text displayed as label for Indigenous languages selector"
  }
}
