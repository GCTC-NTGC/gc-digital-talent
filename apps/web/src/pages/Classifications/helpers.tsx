export function getClassificationLevels(): {
  value: number;
  label: number;
}[] {
  const levels: {
    value: number;
    label: number;
  }[] = [];
  for (let i = 1; i <= 19; i++) {
    levels.push({ value: i, label: i });
  }
  return levels;
}
