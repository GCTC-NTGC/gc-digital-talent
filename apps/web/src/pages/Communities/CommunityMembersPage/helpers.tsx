import orderBy from "lodash/orderBy";
import { IntlShape } from "react-intl";
import EllipsisVerticalIcon from "@heroicons/react/20/solid/EllipsisVerticalIcon";

import { getLocalizedName } from "@gc-digital-talent/i18n";
import { DropdownMenu, IconButton, Link, Ul } from "@gc-digital-talent/ui";
import { notEmpty, unpackMaybes } from "@gc-digital-talent/helpers";
import {
  Maybe,
  Role,
  CommunityMembersPage_CommunityFragment as CommunityMembersPageCommunityFragmentType,
} from "@gc-digital-talent/graphql";

import { CommunityMember } from "~/utils/communityUtils";

import EditCommunityMemberDialog from "./components/EditCommunityMemberDialog";
import RemoveCommunityMemberDialog from "./components/RemoveCommunityMemberDialog";

export function orderRoles(roles: Role[], intl: IntlShape) {
  return orderBy(roles, ({ displayName }) => {
    const value = getLocalizedName(displayName, intl);

    return value
      ? value
          .normalize("NFD")
          .replace(/[\u0300-\u036f]/g, "")
          .toLocaleLowerCase()
      : value;
  });
}

export const actionCell = (
  user: CommunityMember,
  community: CommunityMembersPageCommunityFragmentType,
  hasPlatformAdmin: boolean,
  intl: IntlShape,
) => {
  return (
    <>
      <DropdownMenu.Root>
        <DropdownMenu.Trigger>
          <IconButton
            icon={EllipsisVerticalIcon}
            color="black"
            label={intl.formatMessage(
              {
                defaultMessage: "Actions for {userName} in {communityName}",
                id: "J+haAz",
                description:
                  "Aria label for the menu trigger for community actions",
              },
              {
                userName: `${user.firstName} ${user.lastName}`,
                communityName: getLocalizedName(community.name, intl),
              },
            )}
          />
        </DropdownMenu.Trigger>
        <DropdownMenu.Content>
          <EditCommunityMemberDialog
            user={user}
            community={community}
            hasPlatformAdmin={hasPlatformAdmin}
          />
          <RemoveCommunityMemberDialog
            user={user}
            community={community}
            hasPlatformAdmin={hasPlatformAdmin}
          />
        </DropdownMenu.Content>
      </DropdownMenu.Root>
    </>
  );
};

export function emailLinkCell(
  email: Maybe<string> | undefined,
  intl: IntlShape,
) {
  if (email) {
    return (
      <Link color="black" external href={`mailto:${email}`}>
        {email}
      </Link>
    );
  }

  return (
    <span className="italic">
      {intl.formatMessage({
        defaultMessage: "No email provided",
        id: "1JCjTP",
        description: "Fallback for email value",
      })}
    </span>
  );
}

export function roleCell(roles: Maybe<Maybe<Role>[]>, intl: IntlShape) {
  const nonEmptyRoles = unpackMaybes(roles);
  const roleItems = nonEmptyRoles
    ? orderRoles(nonEmptyRoles, intl).map((role) => (
        <li key={role.id}>{getLocalizedName(role.displayName, intl)}</li>
      ))
    : null;

  return roleItems ? <Ul>{roleItems}</Ul> : null;
}

export function roleAccessor(roles: Maybe<Maybe<Role>[]>, intl: IntlShape) {
  const nonEmptyRoles = roles?.filter(notEmpty);

  return nonEmptyRoles
    ? orderRoles(nonEmptyRoles, intl)
        .map((role) => getLocalizedName(role.displayName, intl))
        .join(", ")
    : "";
}
