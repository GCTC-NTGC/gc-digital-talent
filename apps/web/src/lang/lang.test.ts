import groupBy from "lodash/groupBy"; // provides sorted output
import stringify from "json-stable-stringify";

import * as rawI18nEnMessages from "@gc-digital-talent/i18n/en.json";
import * as rawI18nFrMessages from "@gc-digital-talent/i18n/fr.json";
import { notEmpty } from "@gc-digital-talent/helpers";

import * as rawWebEnMessages from "./en.json";
import * as rawWebFrMessages from "./fr.json";

describe("message files", () => {
  /*
    To update expected changes to the snapshot:
    `npx jest ./src/lang/lang.test.ts --testNamePattern="should have no changes to duplicate strings" --updateSnapshot`
    */
  it("should have no changes to duplicate strings", () => {
    const enMessages: Record<string, { defaultMessage: string }> = {
      ...rawWebEnMessages,
      ...rawI18nEnMessages,
    };

    const frMessages: Record<string, { defaultMessage: string }> = {
      ...rawWebFrMessages,
      ...rawI18nFrMessages,
    };

    const matchedMessages = Object.keys(enMessages).map((messageId) => ({
      en: enMessages[messageId].defaultMessage,
      fr: frMessages[messageId]?.defaultMessage,
    }));
    const groupedByEn = groupBy(matchedMessages, "en");
    const messagesWithMultipleFrStrings = Object.keys(groupedByEn)
      .map((en) => ({
        en,
        fr: Array.from(
          new Set(groupedByEn[en].map((g) => g.fr).filter(notEmpty)),
        ),
      }))
      .filter((g) => g.fr.length > 1);

    expect(
      stringify(messagesWithMultipleFrStrings, { space: "  " }),
    ).toMatchSnapshot();
  });
  it("should have no quotes in french strings", () => {
    const frMessages: Record<string, { defaultMessage: string }> = {
      ...rawWebFrMessages,
      ...rawI18nFrMessages,
    };

    const messagesArr = Object.keys(frMessages).map(
      (messageId) => frMessages[messageId]?.defaultMessage,
    );

    expect(messagesArr).toEqual(
      expect.not.arrayContaining([expect.stringMatching(/"/)]),
    );
  });
});
