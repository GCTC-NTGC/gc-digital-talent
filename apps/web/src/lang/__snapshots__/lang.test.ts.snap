// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message files should have no changes to duplicate strings 1`] = `
"[
  {
    "en": "Membership roles",
    "fr": [
      "Rôles des membres",
      "Rôles dans l'équipe"
    ]
  },
  {
    "en": "Community roles",
    "fr": [
      "Rôles dans la collectivité",
      "Rôles dans des collectivités"
    ]
  },
  {
    "en": "Language requirement",
    "fr": [
      "Exigence linguistique",
      "Exigences relatives à la langue"
    ]
  },
  {
    "en": "Add individual role",
    "fr": [
      "Ajoutez un rôle individuel",
      "Ajouter un rôle individue"
    ]
  },
  {
    "en": "Minimum education",
    "fr": [
      "Niveau d'études minimal",
      "Études minimales"
    ]
  },
  {
    "en": "Remove from process",
    "fr": [
      "Retirez du processus",
      "Retirer du processus"
    ]
  },
  {
    "en": "Remove membership<hidden> in {team}</hidden>",
    "fr": [
      "Retirez de l'équipe<hidden> dans {team}</hidden>",
      "Retirer de l'équipe<hidden> dans {team}</hidden>"
    ]
  },
  {
    "en": "Add team role",
    "fr": [
      "Ajouter un rôle dans l'équipe",
      "Ajoutez un rôle dans l'équipe"
    ]
  },
  {
    "en": "Add process role",
    "fr": [
      "Ajoutez un rôle dans le processus",
      "Ajouter des rôles dans des processus"
    ]
  },
  {
    "en": "Add community role",
    "fr": [
      "Ajouter un rôle dans la collectivité",
      "Ajoutez un rôle dans la collectivité"
    ]
  },
  {
    "en": "Required",
    "fr": [
      "Requises",
      "Champ obligatoire"
    ]
  },
  {
    "en": "Skill requirements",
    "fr": [
      "Exigences relatives aux compétences",
      "Exigences en matière de compétences"
    ]
  },
  {
    "en": "Optional",
    "fr": [
      "Facultatives",
      "Facultatif"
    ]
  },
  {
    "en": "Remove from community",
    "fr": [
      "Retirez de la collectivité",
      "Retirer de la collectivité"
    ]
  },
  {
    "en": "About this role",
    "fr": [
      "À propos de ce rôle",
      "Au sujet de ce rôle"
    ]
  }
]"
`;
